core.ts

```diff
@@ -974,6 +980,56 @@ abstract class PinejsClientCoreTemplate<
 		return this.request(params, { method: 'DELETE' });
 	}
 
+	upsert(params: PinejsClientCoreFactory.UpsertParams) {
+		const { id, body, ...restParams } = params;
+
+		if (!isObject(id)) {
+			throw new Error('The id property must be an object');
+		}
+
+		const naturalKeyProps = mapObj(params.id, (_value, key) => key);
```

Comment:
Why not just use `Object.keys` ?
```suggestion
		const naturalKeyProps = Object.keys(params.id)
```

Reply:
By taking a look at the rest of the code, I which spec level this is assuming to be available.
So, since I saw mapObj as a helper func, I assumed the lower possible and just used it.
I guess that since we are already using `Array.isArray`, we can be almost sure that `Object.keys` is also available...
See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#Browser_compatibility
See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys#Browser_compatibility

PS: I can't stop worrying about such things... But I think that I should...